### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
[Project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#project-requirements)
Create a class Classroom with the following:
@label instance variable, should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
Create the has-many side (a classroom has many students).
Create the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
Create a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
Create a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
Create the has-many side of Book and Rental (a book has many rentals).
Create the belongs-to side of Rental and Book (a rental belongs to a book).
Create the has-many side of Person and Rental (a person has many rentals).
Create the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classes/rental.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :book, :person, :date
+
+  def initialize(date, person = nil, book = nil)
+    @person = person
+    @book = book
+    @date = date
+  end
```
Comment (Davidosky007) [inline classes/rental.rb]: * Kindly add the association `a rental belongs to a book` by adding itself(rental) to the @book achieving the 'belongs-to' association also the same thing for this association `a rental belongs to a person` as [required](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Create%20the%20has%2Dmany%20side%20of,a%20rental%20belongs%20to%20a%20person). Hint below 👇 
 ```ruby
@book = book
    book.rentals << self

    @person = person
   
    person.rentals << self
```

File: classes/student.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(name, age, parent_permission, classroom)
+  attr_accessor :classroom
+
+  def initialize(name, age, parent_permission, classroom = nil)
```
Comment (Davidosky007) [inline classes/student.rb:6]: * Kindly add the association `a student belongs to a classroom` by adding `a method that serves as the 'belongs-to' side (a student belongs to a classroom).`  [ Link to the article with examples](
https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Association between most classes setup properly 👍 
* No linter errors 👍 
* Well-written PR 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/rental.rb (hunk 1 above)
Comment (Abdulwhbhussain) [reply classes/rental.rb]: Implemented

File: classes/student.rb (hunk 2 above)
Comment (Abdulwhbhussain) [reply classes/student.rb:6]: Implemented

**Review by skyv26 — APPROVED** [review]
## Hi @Abdulwhbhussain, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Well Implemented the missing features asked by previous code reviewer regarding associations ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**